### DISCUSSION TITLE
Maximum Text Length Translations

### DISCUSSION THREAD

**Discussion by Alexbuchwald** [JohnSnowLabs/spark-nlp#5742] [Q&A] [answered]
Hi!
I`m new to Spark NLP and I`m working on translation tasks. I noticed that the pretrained models work really well, but unfortunately text will be cut off after a certain length. I was looking through the documentations, but I could not find a refrence or setting for the maximum length. Do you have an idea how to handle very long, multi-sentence long text data?

Thanks in advance! :)

![image](https://user-images.githubusercontent.com/27952718/123693864-e4116900-d858-11eb-8eaf-21d9a0684fc9.png)

**Comment by maziyarpanahi** [comment, marked as answer]
Hi,

You are using NLU: https://github.com/JohnSnowLabs/nlu

In Spark NLP there are 2 parameters to set the input/output lengths. Also, you can feed sentences and which breaks the documents to shorter sentences for better and faster translation.

Perhaps @C-K-Loan can share more info about how to do these in NLU.

**Reply by Alexbuchwald** [reply]
Thanks
